File: src/main/java/ru/dushkinmir/absolutelyRandom/features/sex/SexCommands.java
package ru.dushkinmir.absolutelyRandom.features.sex;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.Argument;
import dev.jorel.commandapi.arguments.PlayerArgument;
import dev.jorel.commandapi.arguments.SafeSuggestions;
import org.bukkit.entity.Player;
import org.bukkit.plugin.Plugin;

public class SexCommands {
    AnalFissureHandler fissureHandler;
    Plugin plugin;

    public SexCommands(AnalFissureHandler fissureHandler, Plugin plugin) {
        this.fissureHandler = fissureHandler;
        this.plugin = plugin;
    }

    public void registerSexCommand() {
        Argument<?> noSelectorSuggestions = new PlayerArgument("target")
                .replaceSafeSuggestions(SafeSuggestions.suggest(info -> {
                    // Получаем игрока, который вводит команду
                    Player senderPlayer = (Player) info.sender();
                    // Получаем всех онлайн игроков, кроме отправителя
                    return plugin.getServer().getOnlinePlayers().stream()
                            .filter(player -> !player.equals(senderPlayer)) // исключаем отправителя
                            .toArray(Player[]::new);
                }));
        new CommandAPICommand("sex")
                .withArguments(noSelectorSuggestions)
                .executes((sender, args) -> {
                    if (sender instanceof Player player) {
                        Player target = (Player) args.get("target");
                        SexManager.triggerSexEvent(player, target, plugin, fissureHandler);
                    }
                })
                .register();
    }
}

File: src/main/java/ru/dushkinmir/absolutelyRandom/features/sex/SexCommands.kt
package ru.dushkinmir.absolutelyRandom.features.sex

import dev.jorel.commandapi.CommandPermission
import dev.jorel.commandapi.kotlindsl.commandTree
import dev.jorel.commandapi.kotlindsl.playerArgument
import dev.jorel.commandapi.kotlindsl.playerExecutor
import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin

class SexCommands(val fissureHandler: AnalFissureHandler, val plugin: Plugin) {

    fun registerSexCommand() {
        commandTree("sex") {
            playerArgument("target") {
                replaceSuggestions { info, builder ->
                    val senderPlayer = info.sender() as Player
                    plugin.server.onlinePlayers.stream()
                        .filter { player -> player != senderPlayer }
                        .forEach { builder.suggest(it.name) }
                    builder.buildFuture()
                }
                withPermission(CommandPermission.NONE)
                playerExecutor { player, args ->
                    val target = args["target"] as Player
                    SexManager.triggerSexEvent(player, target, plugin, fissureHandler)
                }
            }
        }
    }
}